cranelift/codegen/src/machinst/lower.rs

```diff
@@ -122,6 +125,29 @@ pub struct Lower<'func, I: VCodeInst> {
     next_vreg: u32,
 }
 
+/// Notion of "relocation distance". This gives an estimate of how far away a symbol will be from a
+/// reference.
+#[derive(Clone, Copy, Debug, PartialEq, Eq)]
+pub enum RelocDistance {
+    /// Target of relocation is "nearby". The threshold for this is fuzzy but should be interpreted
+    /// as approximately "within the compiled output of one module"; e.g., within AArch64's +/-
+    /// 128MB offset. If unsure, use `Far` instead.
+    Near,
+    /// Target of relocation could be anywhere in the address space.
+    Far,
+}
+
+impl RelocDistance {
+    /// Get a `RelocDistance` from a `colocated` bit in a CLIF instruction's symbol reference.
+    pub fn from_colocated(colocated: bool) -> RelocDistance {
```

Comment:
Could this be a getter on `ExtFuncData` and on `GlobalValueData` instead?

Reply:
Actually not sure how I feel about putting it on `GlobalValueData`, since it should only be for `GlobalVataData::Symbol`, really...

Reply:
Good idea -- done (made the `GlobalValueData` getter return an `Option`, only `Some` for the `Symbol` case).